Gems/Atom/RHI/Code/Include/Atom/RHI/MultiDeviceImage.h

```diff
@@ -145,5 +145,7 @@ namespace AZ::RHI
         const RHI::MultiDeviceImage* m_image;
         //! The corresponding ImageViewDescriptor for this view.
         ImageViewDescriptor m_descriptor;
+        //! SingleDeviceImageView cache
+        AZStd::unordered_map<int, Ptr<RHI::SingleDeviceImageView>> m_cache;
```

Comment:
I forget the details from our earlier discussion but it is not clear why we need this cache. Doesn't every Resource has a m_resourceViewCache which should be caching the views? so a MDImage -> SingleDeviceImage->SingleDeviceImageView

Reply:
We thought so as well, but the issue is that the cache only holds raw pointers:
```cpp
// Cache the resourceViews in order to avoid re-creation
// Since ResourceView has a dependency to Resource this cache holds raw pointers here in order to ensure there
// is no circular dependency between the resource and it's resourceview.
mutable AZStd::unordered_map<size_t, ResourceView*> m_resourceViewCache;
```
This means that the cache only avoids re-creation, but does not "cache" the resource and hold it, this is done at different portions throughout the code-base.
And example would be `RPI::Image`, which holds a `RHI::Ptr<RHI::ImageView> m_imageView;`, i.e. it is holding the memory for the `ImageView`, a pointer to this is then stored in the cache and no further views need to be created.
And creation for the moment works like this 
```cpp
// Ask SingleDeviceImage for an SingleDeviceImageView
Ptr<SingleDeviceImageView> SingleDeviceImage::GetImageView();
// This calls 
Ptr<SingleDeviceImageView> SingleDeviceResource::GetResourceView();
// Which calls (if no view has been created yet)
Ptr<SingleDeviceImageView> SingleDeviceResource::InsertNewImageView();
```
So, in all those instances, the caller is responsible for holding the memory, the `RHI::Ptr` is just passed back to the caller and if this is not captured, the memory will be released again immediately.

Now, let's switch to `MultiDevice*`.
Now, `RPI::Image` holds an `RHI::MultiDeviceImageView` instead of an `RHI::SingleDeviceImageView` and all other places, that previously held `RHI::SingleDeviceImageView` in an `RHI::Ptr` now hold the multi-device variant.
That means that currently no one is holding the memory for the single-device views, which we still need at the moment to pass it to the SRGs for example.
So, we chose for the moment to let the `MultiDeviceImageView` hold the `SingleDeviceImageView`s.
We have not done the conversion of the `FrameGraph` yet, but we hope that we can then remove the cache once again and stay multi-device until the very last moment and then there would be no need to cache it.
I hope that makes sense. :)

Typical code that otherwise fails at the moment would be:
```cpp
m_someSrg->SetImage(m_indexImage, m_Image);
// Which calls internally
const auto* imageView = image ? image->GetImageView() : nullptr; // Here we have a MultiDeviceImageView
if (m_data.SetImageView(inputIndex, imageView->GetDeviceImageView().get(), arrayIndex)) // Here we need to pass in a SingleDeviceImageView (which gets destroyed immediately after the function exits)
...
```